test/path.jl

```diff
@@ -10,6 +10,7 @@ for S in (String, GenericString)
     @test basename(S("foo$(sep)bar")) == "bar"
     @test dirname(S("foo$(sep)bar")) == "foo"
 
+    @test expanduser(S("")) == ""
```

Comment:
There are some `expanduser` tests later in this file too. Not sure if this should go here or later on.